lib/PuppeteerSharp.Tests/PageTests/SetContentTests.cs

```diff
@@ -44,5 +44,20 @@ public async Task ShouldWorkWithHtml4Doctype()
 
             Assert.Equal($"{doctype}{ExpectedOutput}", result);
         }
+
+        [Fact]
+        public async Task ShouldAwaitResourceToLoad()
+        {
+            var imgPath = "/img.png";
+            var imgResponse = new TaskCompletionSource<bool>();
+            Server.SetRoute(imgPath, context => imgResponse.Task);
+            var loaded = false;
+            var contentTask = Page.SetContentAsync($"<img src=\"${TestConstants.ServerUrl + imgPath}\"></img>")
+                .ContinueWith(_ => loaded = true);
+            await Server.WaitForRequest(imgPath);
```

Comment:
We might have some race conditions here. I'd follow this pattern.

Reply:
what pattern?

Reply:
not sure what to do about this test 😢